src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -214,6 +219,17 @@ void ErrorHandler(object sender, Newtonsoft.Json.Serialization.ErrorEventArgs ev
                         }
                     }
 
+                    if (SendBadRequestForExceptionsDuringDeserialization
+                        && (exception is JsonException || exception is OverflowException))
+                    {
+                        throw new InputFormatException("Error deserializing input", exception);
+                    }
+                    else
+                    {
+                        var exceptionDispatchInfo = ExceptionDispatchInfo.Capture(exception);
+                        exceptionDispatchInfo.Throw();
+                    }
```

Comment:
We specifically don't want to turn these kinds of errors into exceptions. **Ever**